lib/src/number_increment_decrement.dart

```diff
@@ -660,30 +678,47 @@ class _NumberInputWithIncrementDecrementState
             Expanded(
               flex: 1,
               child: TextFormField(
-                validator: widget.validator ?? _minMaxValidator,
-                style: widget.style,
-                enabled: widget.enabled,
-                textAlign: TextAlign.center,
-                autovalidate: widget.autovalidate,
-                decoration: widget.numberFieldDecoration ??
-                    InputDecoration(
-                      border: OutlineInputBorder(
-                        borderRadius: BorderRadius.circular(5.0),
+                  validator: widget.validator ?? _minMaxValidator,
+                  style: widget.style,
+                  enabled: widget.enabled,
+                  textAlign: TextAlign.center,
+                  autovalidate: widget.autovalidate,
+                  decoration: widget.numberFieldDecoration ??
+                      InputDecoration(
+                        border: OutlineInputBorder(
+                          borderRadius: BorderRadius.circular(5.0),
+                        ),
                       ),
-                    ),
-                controller: _controller,
-                keyboardType: TextInputType.numberWithOptions(
-                  decimal: !widget.isInt,
-                  signed: true,
-                ),
-                inputFormatters: <TextInputFormatter>[
-                  widget.isInt
-                      ? FilteringTextInputFormatter.digitsOnly
-                      : FilteringTextInputFormatter.allow(
-                          RegExp("[0-9.]"),
-                        )
-                ],
-              ),
+                  controller: _controller,
+                  keyboardType: TextInputType.numberWithOptions(
+                    decimal: !widget.isInt,
+                    signed: true,
+                  ),
+                  inputFormatters: <TextInputFormatter>[
+                    widget.isInt
+                        ? FilteringTextInputFormatter.digitsOnly
+                        : FilteringTextInputFormatter.allow(
+                            RegExp("[0-9.]"),
+                          )
+                  ],
+                  onFieldSubmitted: (value) {
+                    if (this.widget.onSubmitted != null) {
+                      num newVal;
+                      try {
+                        newVal = this.widget.isInt
+                            ? int.parse(value)
+                            : double.parse(value);
+                      } catch (e) {
+                        print("cannot convert $value into a number, e=$e");
```

Comment:
If you don't need this print statement, shall I remove it. I am not a big fan of print statements in code.